ca_gazebo/src/virtual_wall_detector.cpp

```diff
@@ -0,0 +1,75 @@
+#include "ca_gazebo/virtual_wall_detector.h"
+
+#include <gazebo/common/Plugin.hh>
+
+static const char create2_model_name_prefix[] = "irobot_create2.";
+static const size_t create2_model_name_prefix_length = sizeof(create2_model_name_prefix) - 1;
+
+namespace gazebo
+{
+  // Register this plugin with the simulator
+  GZ_REGISTER_MODEL_PLUGIN(GazeboVirtualWallDetector)
+
+  void GazeboVirtualWallDetector::Load(physics::ModelPtr _parent, sdf::ElementPtr _sdf)
+  {
+    // Make sure the ROS node for Gazebo has already been initialized
+    if (!ros::isInitialized())
+    {
+      ROS_FATAL_STREAM("A ROS node for Gazebo has not been initialized, unable to load plugin. "
+        << "Load the Gazebo system plugin 'libgazebo_ros_api_plugin.so' in the gazebo_ros package)");
+      return;
+    }
+    if (!ros::isInitialized()) 
+    {
+      int argc = 0;
+      char** argv = NULL;
+      ros::init(argc, argv, "virtual_wall_detector");
+    }
+    // Initialize the ros variables and gaebo variables
+    const std::string & name = _parent->GetName();
+    std::istringstream iss(name.substr(create2_model_name_prefix_length));
+    size_t x;
+    iss >> x;
+//    ROS_WARN("The name %s",name.c_str());
+    if (iss && iss.eof()) 
+    {
+      std::ostringstream oss;
+      oss << "create" << x << "/virtual_wall/raw/";
+      std::string topic_name_ = oss.str();
```

Comment:
Local variables must not have an ending `_`. **Be consistent with the coding style.**